### DISCUSSION TITLE
What are the recommendation for DB connection string having @ symbol in their database connection values

### DISCUSSION THREAD

**Discussion by soujanyat** [sqlalchemy/sqlalchemy#10981] [Usage Questions]
SQLAlchemy = 2.0.25
z3c.sqlalchemy = 2.1.1 
zope.sqlalchemy = 3.1 

we have database connection string which contains '@' symbol in their user_name and password string.

The after parsing from parse_ulr() call, invalid string is assigning for DSN_NAME which lead to DataSourceName  not found error.

With the latest version of SQLAlchemy, encounter this error,  as it is working with 1.3.17 version.

What is the alternative suggestion for this scenario

Thank You,

**Comment by CaselIT** [comment]
Hi,

Just follow the documentation: https://docs.sqlalchemy.org/en/20/core/engines.html#database-urls
